osu.Game.Rulesets.Catch/Skinning/LegacyFruitPiece.cs

```diff
@@ -51,7 +52,7 @@ private void load(DrawableHitObject drawableObject, ISkinSource skin)
                 },
             };
 
-            if (drawableCatchObject.HitObject.HyperDash)
+            if (((PalpableCatchHitObject)drawableCatchObject.HitObject).HyperDash)
```

Comment:
osu! has this pattern where resolved DHOs are cast to the more-specific type apt to the drawable piece at play and stored to fields with that type which I quite like and would make this a bit less of an eyesore:

https://github.com/ppy/osu/blob/7f3c8ad744df80079bc377b63c570ca6243426f9/osu.Game.Rulesets.Osu/Objects/Drawables/Pieces/DefaultSpinnerDisc.cs#L51-L54